Security/Authorization/Expression/ExpressionParser.php

```diff
@@ -105,6 +105,10 @@ private function Primary()
             return new ConstantExpression($this->match(ExpressionLexer::T_STRING));
         }
 
+        if ($this->lexer->isNext(ExpressionLexer::T_INTEGER)) {
+            return new ConstantExpression((int) $this->match(ExpressionLexer::T_INTEGER));
```

Comment:
casting the value to an integer should rather be done in the lexer, to be consistent with the handling of strings